src/contexts/auth.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { View, ActivityIndicator } from "react-native";
+import * as auth from '../services/Auth/auth'
+
+interface AuthContextData {
+  signed: boolean,
+  user: object | null,
+  loading: boolean,
+  signIn(email: string, password: string): Promise<void>,
+  signOut(): void
+}
+
+const AuthContext = createContext<AuthContextData>({} as AuthContextData)
+
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState<object | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
+
+  // useEffect(() => {
+  //   async function loadStorageData() {
+  //     const storagedUser = await AsyncStorage.getItem('@SETAuth:user')
+  //     if (storagedUser) {
+  //       setUser(JSON.parse(storagedUser))
+  //       setLoading(false)
+  //     }
+
+  //   }
+  //   loadStorageData()
+  // }, [])
+
+  async function signIn(email: string, password: string): Promise<void> {
+    try {
+      const response = await auth.signIn(email, password)
+      setUser(response.user)
+      // await AsyncStorage.setItem('@SETAuth:user', JSON.stringify(response.user))
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏we could open a new issue in this repo to add the `AsyncStorage` functionality and remove this comments too